test/Microsoft.SqlTools.ServiceLayer.Test/LanguageServer/SqlCompletionItemTests.cs

```diff
@@ -0,0 +1,179 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Microsoft.SqlServer.Management.SqlParser.Intellisense;
+using Microsoft.SqlTools.ServiceLayer.LanguageServices;
+using Microsoft.SqlTools.ServiceLayer.LanguageServices.Contracts;
+using Xunit;
+
+namespace Microsoft.SqlTools.ServiceLayer.Test.LanguageServer
+{
+    public class SqlCompletionItemTests
+    {
+        [Fact]
+        public void InsertTextShouldIncludeBracketGivenNameWithSpace()
+        {
+            string declarationTitle = "name with space";
+            string expected = "[" + declarationTitle + "]";
+
+            DeclarationType declarationType = DeclarationType.Table;
+            string tokenText = "";
+            SqlCompletionItem item = new SqlCompletionItem(declarationTitle, declarationType, tokenText);
+            CompletionItem completionItem = item.CreateCompletionItem(0, 1, 2);
+
+            Assert.True(completionItem.InsertText.StartsWith("[") && completionItem.InsertText.EndsWith("]"));
+        }
+
+        [Fact]
+        public void InsertTextShouldIncludeBracketGivenNameWithSpecialCharacter()
+        {
+            string declarationTitle = "name @";
+            string expected = "[" + declarationTitle + "]";
+            DeclarationType declarationType = DeclarationType.Table;
+            string tokenText = "";
+            SqlCompletionItem item = new SqlCompletionItem(declarationTitle, declarationType, tokenText);
+            CompletionItem completionItem = item.CreateCompletionItem(0, 1, 2);
+
+            Assert.Equal(completionItem.InsertText, expected);
+            Assert.Equal(completionItem.Detail, declarationTitle);
+            Assert.Equal(completionItem.Label, declarationTitle);
+        }
+
+        [Fact]
+        public void LabelShouldIncludeBracketGivenTokenWithBracket()
+        {
+            string declarationTitle = "name";
+            string expected = "[" + declarationTitle + "]";
+            DeclarationType declarationType = DeclarationType.Table;
+            string tokenText = "[";
+            SqlCompletionItem item = new SqlCompletionItem(declarationTitle, declarationType, tokenText);
+            CompletionItem completionItem = item.CreateCompletionItem(0, 1, 2);
+
+            Assert.Equal(completionItem.Label, expected);
+            Assert.Equal(completionItem.InsertText, expected);
+            Assert.Equal(completionItem.Detail, expected);
+        }
+
+        [Fact]
+        public void LabelShouldIncludeBracketGivenTokenWithBrackets()
```

Comment:
I think we should have a test covering escaped brackets. E.g. what happens in the following case? Will auto-complete work correctly? So long as this passes I'm happy to sign off

```

-- Create the table in the specified schema
CREATE TABLE [Bracket\[]
(
    C1 INT NOT NULL PRIMARY KEY, -- primary key column
    C2 [NVARCHAR](50) NOT NULL,
    C3 [NVARCHAR](50) NOT NULL
   -- specify more columns here
);
GO
select C1 from dbo.[Bracket\[] 
```